lib/OpenQA/Task/Needle/Delete.pm

```diff
@@ -18,7 +18,7 @@ sub _delete_needles ($app, $minion_job, $args) {
     # SignalGuard will prevent the delete task to interrupt with no recovery,
     # instead will retry once the gru server returned up and running. The popup
     # on the frontend will wait until the retried job finished.
-    my $signal_guard = OpenQA::Task::SignalGuard->new($minion_job);
+    my $signal_guard = OpenQA::Task::SignalGuard->new($minion_job)->abort(1);
```

Comment:
The following is supposed to work as well and should be faster (minimizing the race condition we have here):
```suggestion
    my $signal_guard = OpenQA::Task::SignalGuard->new($minion_job, abort => 1);
```